bayes_opt/bayesian_optimization.py

```diff
@@ -18,30 +19,52 @@
 from bayes_opt.target_space import TargetSpace
 from bayes_opt.util import ensure_rng
 
+if TYPE_CHECKING:
+    from collections.abc import Callable, Iterable, Mapping, Sequence
+
+    import numpy as np
+    from numpy.typing import NDArray
+
+    from bayes_opt.acquisition import AcquisitionFunction
+    from bayes_opt.constraint import ConstraintModel
+    from bayes_opt.domain_reduction import DomainTransformer
+
+    class Constraint(Protocol):
```

Comment:
what's the argument behind not using `scipy.optimize.NonlinearConstraint`?

Reply:
If we use `NonlinearConstraint` directly, it will only accept `NonlinearConstraint`.
If we use `Protocol`, it accepts `ConstraintModel`.

Reply:
I had a look and I understand now. I think that docstring is false, it should not accept a `ConstraintModel`, it should just accept the `NonLinearConstraint`.

Reply:
But it's being used here.
https://github.com/bayesian-optimization/BayesianOptimization/blob/2298f7c5c9c7b6c9b64669bacd0e87f9ba911f96/bayes_opt/bayesian_optimization.py#L133-L142

Reply:
here a `scipy.optimize.NonLinearConstraint` object is used to create a `ConstraintModel`, right? if `constraint` was already a `ConstraintModel`, one could just `constraint_ = constraint` without constructing a new one.

Reply:
I agree. I was confused about `BayesianOptimization` and `TargetSpace`, I'll fix it.

Reply:
i fixed it